repose-aggregator/commons/commons-utilities/src/test/scala/org/openrepose/commons/utils/servlet/http/HttpServletResponseWrapperTest.scala

```diff
@@ -2003,5 +2013,137 @@ class HttpServletResponseWrapperTest extends FunSpec with BeforeAndAfter with Ma
       mockResponse.isCommitted shouldBe true
       wrappedResponse.isCommitted shouldBe true
     }
+
+    it("should write the content-length header") {
+      val out = new ByteArrayServletOutputStream()
+      val wrappedResponse = new HttpServletResponseWrapper(originalResponse, ResponseMode.PASSTHROUGH, ResponseMode.MUTABLE, out)
+
+      val body = "foo"
+      wrappedResponse.getOutputStream.print(body)
+
+      wrappedResponse.commitToResponse()
+
+      originalResponse.getHeader(CommonHttpHeader.CONTENT_LENGTH.toString) shouldEqual "3"
+    }
+
+    it("should mark the wrapped response as committed") {
+      val wrappedResponse = new HttpServletResponseWrapper(originalResponse, ResponseMode.PASSTHROUGH, ResponseMode.PASSTHROUGH)
+
+      wrappedResponse.commitToResponse()
+
+      wrappedResponse.isCommitted shouldBe true
+    }
+  }
+
+  describe("setStatus") {
+    it("should set the status code and message") {
+      val wrappedResponse = new HttpServletResponseWrapper(originalResponse, ResponseMode.PASSTHROUGH, ResponseMode.PASSTHROUGH)
+
+      wrappedResponse.setStatus(418, "TEAPOT")
+
+      wrappedResponse.getStatus shouldBe 418
+      wrappedResponse.getMessage shouldEqual "TEAPOT"
+    }
+  }
+
+  describe("sendError") {
+    it("should set the status code and message") {
+      val wrappedResponse = new HttpServletResponseWrapper(originalResponse, ResponseMode.PASSTHROUGH, ResponseMode.PASSTHROUGH)
+
+      // A hack to make this test work -- the original response /should/ set the status when sendError is called, but
+      // this mock doesn't. So I set it here. If sendError overwrites it like it should, great, if not, fine.
+      originalResponse.setStatus(418)
+
+      wrappedResponse.sendError(418, "TEAPOT")
+
+      wrappedResponse.getStatus shouldBe 418
+      wrappedResponse.getMessage shouldEqual "TEAPOT"
+    }
+
+    it("should mark the wrapped as committed when given a single argument") {
+      val wrappedResponse = new HttpServletResponseWrapper(originalResponse, ResponseMode.PASSTHROUGH, ResponseMode.PASSTHROUGH)
+
+      wrappedResponse.sendError(418)
+
+      wrappedResponse.isCommitted shouldBe true
+    }
+
+    it("should mark the wrapper as committed when given two arguments") {
+      val wrappedResponse = new HttpServletResponseWrapper(originalResponse, ResponseMode.PASSTHROUGH, ResponseMode.PASSTHROUGH)
+
+      wrappedResponse.sendError(418, "TEAPOT")
+
+      wrappedResponse.isCommitted shouldBe true
+    }
+
+    it("should not commit the underlying response if not in a mutable header mode") {
```

Comment:
Copy pasta'd the name. Apparently I changed the negation on the mode, but not on the commit. Will fix.